examples/pytorch/gin/model.py

```diff
@@ -0,0 +1,300 @@
+            pooled[node_id] = sum_features
+
+        pooled_rep = self.mlps[layer](pooled)
+        h = self.batch_norms[layer](pooled_rep)
+        h = F.relu(h)
+        return h
+
+    def get_score(self, context: dict):
+        num_nodes = len(context["nb_counts"][0])
+        nb_counts = context["nb_counts"].squeeze()
+        features = context["features"].squeeze()
+        nb_features = context["nb-features"].squeeze()
+
+        # graph_pool = self.graphpool(nb_features, nb_counts, num_nodes)
+        # get_logger().info("NB COUNTS: " + str(nb_counts))
+
+        hidden_rep = [features]
+        h = features
+
+        for layer in range(self.num_layers - 1):
+            h = self.next_layer(h, layer, nb_counts, num_nodes)
+            hidden_rep.append(h)
+
+        score = 0
+
+        for layer, h in enumerate(hidden_rep):
+
+            # get_logger().info("Layer " + str(layer) + " =====================================================")
+            ma = MeanAggregator(h)
+            pooled_h = ma.forward(h, num_nodes)
+
+            # get_logger().info("Pooled h " + str(layer)
+
+            # get_logger().info("Pooled h shape: " + str(pooled_h.shape))
+            score += F.dropout(
+                self.linears_prediction[layer](pooled_h),
+                self.final_dropout,
+                training=self.training,
+            )
+            # get_logger().info("==========================================================================================")
+
+        return score
+
+    def forward(self, context: dict):
+        scores: torch.Tensor = self.get_score(context)
+        # get_logger().info("Scores: " )
+        labels = context["label"].long().squeeze().clone().detach()
+
+        loss = self.xent(scores, labels)
+
+        # Take argmax to fetch class indices
+        scores = scores.argmax(dim=1)
+        # get_logger().info("Scores: " + str(scores.float().mean()))
+
+        return (loss, scores, labels)
+
+    def metric_name(self):
+        """Metric used for model evaluation."""
+        return self.metric.name()
+
+    def query(self, graph: Graph, inputs: np.array):
+        """Fetch training data from graph."""
+        context = {"inputs": inputs}
+
+        # get_logger().info("USING SAMPLE NEIGBORS *****************************")
+        # context['neighbors'] = graph.sample_neighbors(
+        #     nodes = context['inputs'],
+        #     edge_types = np.array(self.edge_type),
+        #     count = 67,
+        #     strategy = "randomwithoutreplacement"
+        # )[0]
+
+        # get_logger().info("LEN 1: " + str(len(context['neighbors'])))
+        # get_logger().info("**********************************************************")
+
+        # get_logger().info("USING NORMAL NEIGBORS *****************************")
+        context["neighbors"] = graph.neighbors(
+            context["inputs"], np.array(self.edge_type)
+        )[0][: len(context["inputs"])]
```

Comment:
I think this slice is not right, I think you are dropping a lot of neighbors? If you just want say 10 neighbors per node you should use uniform sample neighobrs instead of this which is full neighbors.